### DISCUSSION TITLE
Need code to login to SharePoint using Console App in Application Permission using Client ID and Client Secret & not using Client Certificates.

### DISCUSSION THREAD

**Discussion by R-LIBIN** [pnp/pnpcore#726] [Q&A]
No User Interaction and to access files.   

Need code Upload large files. It fails deliberately. Please help     us.

**Comment by jansenbe** [comment]
@R-LIBIN : PnP Core SDK uses REST and Graph and both need to be configured in your Azure AD app. Problem is that SharePoint REST APIs do require certificate based authentication when using application permissions, there's no way around that. For a 100% Microsoft Graph based approach you can work with client id and secret. Given the PnP Core large file upload uses REST calls you'll have to use certificate based auth.

**Comment by R-LIBIN** [comment]
We will check bin it. We will try to get certificate.    Thanks a lot from
for that. So do that mean certificate from app?

If we go through graph, is there any way out to send large files to
SharePoint.

I find only OneDriveItem for that.

Kindly help.

Looks to be little bit of work around.

I tried self signed cert but was unsuccessful. I says

Thank you.

This is the try for self signed.

After adding this. in the .csproj. correct me if am wrong. Using this did.

URL : https://github.com/aspnet/DotNetTools/issues/169
  # Ref


PS C:\Users\Administrator\source\repos\SharePointPnP> dotnet user-secrets
-p SharePointPnP.csproj set "Kestrel:Certificates:Development:Password"
"<client secret code>"
System.InvalidOperationException: Invalid character '/' found in the user
secrets ID at index '38'.
   at
Microsoft.Extensions.Configuration.UserSecrets.PathHelper.GetSecretsPathFromSecretsId(String
userSecretsId)
   at
Microsoft.Extensions.SecretManager.Tools.Internal.SecretsStore..ctor(String
userSecretsId, IReporter reporter)
   at Microsoft.Extensions.SecretManager.Tools.Program.RunInternal(String[]
args)
   at Microsoft.Extensions.SecretManager.Tools.Program.TryRun(String[]
args, Int32& returnCode)
Command failed : Invalid character '/' found in the user secrets ID at
index '38'.

<PropertyGroup>
       <UserSecretsId>{client-secret}</UserSecretsId>
    </PropertyGroup>

Regards  ,

Libin

**Reply by jansenbe** [reply]
See https://stackoverflow.com/a/59776567 for using a Graph SDK only approach

**Comment by R-LIBIN** [comment]
Thanks a lot.

**Comment by R-LIBIN** [comment]
But, It seems its for one-drive. Noe, all fine but how we create cert is it
organization generated or self-signed. I got little messed up. I got
confused. Kindly help.

And if self signed where should self signed client system or
ognisation end.

If at all self signed. Can that be signed from any system and used
everywhere or only in the system we pertain to use.

**Reply by jansenbe** [reply]
Using a self signed certificate is fine, we do the same in our samples. Here can see how to use PnP PowerShell to create an Azure AD configured with a newly created self signed certificate: https://pnp.github.io/pnpcore/demos/Demo.AzureFunction.OutOfProcess.AppOnly/readme.html#create-and-configure-the-azure-ad-application
